flink-libraries/flink-table/src/test/scala/org/apache/flink/api/scala/batch/sql/CalcITCase.scala

```diff
@@ -270,6 +271,24 @@ class CalcITCase(
   }
 
   @Test
+  def testNotInWithNestedSelect(): Unit = {
+    val env = ExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env, config)
+    tEnv.getConfig.setNullCheck(true)
+
+    val ds = CollectionDataSets.getSmall3TupleDataSet(env)
+    val table = ds.toTable(tEnv).as('a1, 'a2, 'a3)
+    tEnv.registerTable("A", table)
+    val excludedVal = 1
+    val sqlQuery = s"SELECT a1 FROM A WHERE a1 NOT IN (SELECT a1 FROM A WHERE a1 = $excludedVal)"
+
+    val expectedCnt = ds.collect.count(_._1 != excludedVal)
+    val rowsCnt = tEnv.sql(sqlQuery).count
```

Comment:
Can we check the exact result here? ITCases are quite heavy, so it makes sense to make the check as precise as possible.